### DISCUSSION TITLE
Using signals with Nano JSX

### DISCUSSION THREAD

**Discussion by adwinmbd** [nanojsx/nano#134] [Help / Questions] [answered]
Hello,
I just discovered Nano JSX and find it really interesting. 
I was wondering if it is possible to use [@preact/signals-core](https://github.com/preactjs/signals) with Nano JSX.
Using the [codesandbox](https://nanojsx.io) in the project homepage I came up with this snippet:
```
import {signal} from "https://esm.sh/*@preact/signals-core@1.2.3";

const Nano = nanoJSX;
const { h } = Nano;

const count = signal(0);

const changeValue = (val) => {
  count.value += val;
  console.log(count.value);
};

const App = () => {
    return (
    <div id="counter">
      <span>Counter: {count.value}</span>
      <div>
        <button onClick={() => changeValue(1)}>Increment</button>
        <button onClick={() => changeValue(-1)}>Decrement</button>
      </div>
    </div>
  );
};

Nano.render(<App name="Nano" />, document.body);
```
When the page is loaded `count.value` is at 0. When clicking on the buttons its value is still stuck at 0. There is no update to the rendered count.value. However, when using `console.log` the count.value is being changed at each click.
The same behavior happens with class components.

```
import {signal} from "https://esm.sh/*@preact/signals-core@1.2.3";

const Nano = nanoJSX;
const { h } = Nano;

const count = signal(0);

export class Signo extends Component {
  value;

  constructor(props) {
    super(props);
    this.id = "signo-comp";
    this.value = count.value;
  }

  changeValue(newValue) {
    count.value += newValue;
    console.log(count.value);
    this.update();
  }

  render() {
    return (
      <div id="signo">
        <div>Signo: {this.value}</div>
        <button onClick={() => this.changeValue(1)}>Increment</button>
        <button onClick={() => this.changeValue(-1)}>Decrement</button>
      </div>
    );
  }
}

Nano.render(<App name="Nano" />, document.body);
```

Does anyone know how to solve the issue of the value not being updated when signals value changes ?
Thanks .

**Comment by yandeu** [comment, marked as answer]
Nano JSX is not Preact compatible. Both work completely different. There is no such thing as Signals in NanoJSX.

**Reply by adwinmbd** [reply]
I understand.
Thank you for your good work on Nano JSX.
